lib/common/actions/index.js

```diff
@@ -49,14 +49,21 @@ export function secureFetch (url: string, method: string = 'get', payload?: any,
       .then(res => {
         // if raw response is requested
         if (raw) return res
-        let action, message
+        let action
         // check for errors
         const {status} = res
         if (status >= 500) {
           action = 'RELOAD'
-          message = `Network error!\n\n(${method} request on ${url})`
-          consoleError(message)
-          dispatch(setErrorMessage({message, action}))
+          res.json().then(json => {
```

Comment:
How about errors where the server does not return JSON content?

Reply:
@binh-dam-ibigroup What info do we have if the server does not return JSON content? Should this just be handled in the promise failureCallback as the previous error? Something like: 
`res.json().then(
    json => {... as in this PR ...},
    () => { message = `Network error!\n\n(${method} request on ${url})`
        consoleError(message)
        dispatch(setErrorMessage({message, action}))
    }
)
`

Reply:
@philip-cline, there's a lot of repeated code here from the 400 status conditional block. I'm going to share some changes that [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) things out a bit